streamly.cabal

```diff
@@ -319,7 +319,7 @@ common compile-options
       cpp-options:    -DCABAL_OS_WINDOWS
 
     if flag(streamk)
-      cpp-options:    -DUSE_STREAMK_ONLY
+      cpp-options:    -DDISABLE_FUSION
```

Comment:
1. There is already a `no-fusion` flag in streamly-core. So let's use that flag for fusion disabled build instead of using streamk.
2. We do not need to define the DISABLE_FUSION macro in the streamly package, as it is not used anywhere, it is only used in the streamly-core package.
3. Remove the streamk flag from streamly as well as streamly-core packages.
4. Remove streamk build from CI and use the no-fusion flag build instead.

Reply:
> Remove streamk build from CI and use the no-fusion flag build instead.

This is not take care of yet.